### PR TITLE
Fix filter demo carrier

### LINKED ISSUE #25800: [BO][Carriers] Can't not filter by the first demo carrier (with id=1)
#### Describe the bug
Cannot find default carriers when filtering by its name

#### Expected behavior
Carriers displayed

#### Steps to Reproduce

1. Fresh install shop
2. Go to BO > Carriers page
3. Filter By the name of the first carrier on the list (id=1) (the name should contain 0)
4. See error : Table is empty
5. Click on Edit of the first carrier
6. Just click on Save
7. Back to the list
8. Filter by name => the first carrier is well displayed

**Screenshots**
image
image

#### Additional information

* PrestaShop version: 1.7.6.7 & 1.7.7.7 & 1.7.8.0
* PHP version: 7.3

### PR DESCRIPTION
**Pull request by ghost**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Cannot find default carriers when filtering by its name for the first  carrier created by the installation
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | yes
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/25800
| How to test?      | See issue
| Possible impacts? | 

**:notebook: BC break** 

If a merchant never update the demo carrier, he will see 0
Remove : 

- `Carrier::getCarrierNameFromShopName`
- `Cart::replaceZeroByShopName`
- AdminCarrierController : `getList`

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/26071)

### CODE REVIEW

**Review by matks — COMMENTED** [review]
Should the name be "0", "PrestaShop" or "Example carrier"?

**Review by sowbiba — COMMENTED** [review]
I read it 4 times. I don't get it

In the xml, there was 0 previously, in the screenshot you gave, I see Prestashop as carrier name. You are trying to filter the column with '0' (zero) in the value

And in your PR I see 'Example carrier'

Maybe we need some explanations here to do the review

**Comment by ghost** [comment]
Hello @matks,

This is not specified on the  issue

0 is the installation
Prestashop is the name of the shop if the shop is Toto you will see Toto
Carrier Example is to replace the 0

@sowbiba sowbiba

It's not a problem that I opened, Did you understand better with the comment above

Otherwise let's ask @boubkerbribri

**Comment by kpodemski** [comment]
This doesn't make any sense. We are setting `Example carrier` in the installation data to filter by `Prestashop` name. What happens if we have `0` for the name of the carrier?

I remember some similar issue with the default stores data 😂 But then it was because of the space before the name of the store, crazy 😂

**Comment by ghost** [comment]
_We are setting Example carrier in the installation data to filter by Prestashop name_

I don't want filter PrestaShop :) hard to explain

ScreenShot Tool -20211009235255

ScreenShot Tool -20211009235444

ScreenShot Tool -20211009235607

I dont understand this : 

https://github.com/PrestaShop/PrestaShop/blob/develop/controllers/admin/AdminCarriersController.php#L631-L655

We can remove the method with a good instalation value

It's a very very old feature

https://github.com/PrestaShop/PrestaShop/commit/cc65b4e0cb7c1b3b7bf8ac07f09fd7669b96bf74#diff-9cc4f715b26cedae04644d9ad82357494906498542fbc615f2e1f21bc4b3556e

https://github.com/PrestaShop/PrestaShop/pull/25302

**Comment by ghost** [comment]
More code : 

https://github.com/PrestaShop/PrestaShop/blob/develop/controllers/admin/AdminCartsController.php#L911-L914

https://github.com/PrestaShop/PrestaShop/blob/develop/classes/Carrier.php#L584-L588

https://github.com/PrestaShop/PrestaShop/blob/develop/classes/Carrier.php#L765

https://github.com/PrestaShop/PrestaShop/blob/develop/classes/Carrier.php#L196-L198

https://github.com/PrestaShop/PrestaShop/blob/develop/classes/Carrier.php#L1704-L1718

....

**Comment by ghost** [comment]
@matks @Progi1984 @kpodemski @sowbiba 

can you confirm that all codes with carrier.name == 0 are no longer useful if at the installation we have a good name.

@marionf  

What name could we use ?

**Comment by Progi1984** [comment]
@okom3pom I think so. I think it's a bad idea to have used a `0` as replacement value.

**Comment by ghost** [comment]
I think this is a technical debt that we should remove for the 8.0, no merchant uses the name of his shop for a carrier ( Store pickup , Warehouse pickup ... ) but not the name of the shop

And for the 8.0 we already have two opened issue because of this

**Comment by marionf** [comment]
Hello,

> What name could we use ?

We talked about that with @Julievrz and we suggest "Click and collect"

**Review by matks — COMMENTED** [review]
Hello @okom3pom , @marionf suggested a new (better) name. Wdyt ?

**Comment by ghost** [comment]
Hello @matks && @marionf 
I'm going to focus on all my PR in the next week.
Currently it's a rush for e-commerce sites.
Thomas

**Comment by matks** [comment]
> Hello @matks && @marionf I'm going to focus on all my PR in the next week. Currently it's a rush for e-commerce sites. Thomas

Happy rush, happy benefits and come back when you're rested 😘

File: classes/order/Order.php [hunk 1]
```diff
@@ -1988,9 +1988,6 @@ public function getShipping()
             WHERE o.`id_order` = ' . (int) $this->id . '
             GROUP BY c.id_carrier'
         );
-        foreach ($results as &$row) {
-            $row['carrier_name'] = Cart::replaceZeroByShopName($row['carrier_name'], null);
-        }
 
         return $results;
```
Comment (Progi1984) [inline classes/order/Order.php]: BEtter you can directly return the `Db::getInstance()->executeS`

**Review by Progi1984 — COMMENTED** [review]
A small feedback

**Review by Progi1984 — DISMISSED** [review]

**Comment by ghost** [comment]
I will open a news one

**Pull request closed without merging**